In [15]:

from keys import FTX_API_KEY, FTX_API_SECRET
from datetime import datetime
import time
import hmac
import pandas as pd
from requests import Request, Session, Response
from typing import Optional, Dict, Any, List
from utility_classes.historical_data_processor import FTXDataProcessor
import dateutil.parser as dp




## all non-expired futures tickers

In [2]:
import json

acc = FTXDataProcessor(api_key=FTX_API_KEY, api_secret=FTX_API_SECRET)
current_futures = acc.get_all_futures_tickers()
for i in current_futures:
    print(i)


1INCH-PERP
1INCH-0326
1INCH-0625
AAPL-0326
AAPL-0625
AAVE-PERP
AAVE-0326
AAVE-0625
ABNB-0326
ABNB-0625
ACB-0326
ACB-0625
ADA-PERP
ADA-0326
ADA-0625
ALGO-PERP
ALGO-0326
ALGO-0625
ALPHA-PERP
ALT-PERP
ALT-0326
ALT-0625
AMC-0326
AMC-0625
AMD-0326
AMD-0625
AMPL-PERP
AMZN-0326
AMZN-0625
APHA-0326
APHA-0625
AR-PERP
ARKK-0326
ARKK-0625
ATOM-PERP
ATOM-0326
ATOM-0625
AVAX-PERP
AVAX-0326
AVAX-0625
BABA-0326
BABA-0625
BADGER-PERP
BAL-PERP
BAL-0326
BAL-0625
BAND-PERP
BAO-PERP
BAT-PERP
BB-0326
BB-0625
BCH-PERP
BCH-0326
BCH-0625
BILI-0326
BILI-0625
BITW-0326
BITW-0625
BNB-PERP
BNB-0326
BNB-0625
BNT-PERP
BNTX-0326
BNTX-0625
BRZ-PERP
BRZ-0326
BRZ-0625
BSV-PERP
BSV-0326
BSV-0625
BTC-PERP
BTC-MOVE-0321
BTC-MOVE-0322
BTC-0326
BTC-MOVE-2021Q1
BTC-MOVE-WK-0326
BTC-MOVE-WK-0402
BTC-MOVE-WK-0409
BTC-MOVE-WK-0416
BTC-0625
BTC-MOVE-2021Q2
BTC-0924
BTC-MOVE-2021Q3
BTC-1231
BTC-HASH-2021Q1
BTMX-PERP
BTMX-0326
BTMX-0625
BTT-PERP
BYND-0326
BYND-0625
CAKE-PERP
CEL-0625
CGC-0326
CGC-0625
CHZ-PERP
CHZ-0326
CHZ-0625
CO

## getting all expired futures

In [3]:
expired_futures = acc._get('expired_futures')
for ticker in expired_futures: 
    if ticker['underlying'] == 'ETH':
        print(ticker['name'], ticker['expiryDescription'], ticker['expiry'])


ETH-1225 December 2020 2020-12-25T03:00:00+00:00
ETH-0925 September 2020 2020-09-25T03:00:00+00:00
ETH-0626 June 2020 2020-06-26T03:00:00+00:00
ETH-20200327 March 2020 2020-03-27T03:00:00+00:00
ETH-20191227 December 2019 2019-12-27T03:00:00+00:00
ETH-20190927 September 2019 2019-09-27T03:00:00+00:00
ETH-20190628 June 2019 2019-06-28T03:00:00+00:00
ETH-20190329 March 2019 2019-03-29T03:00:00+00:00


## all past futures expiration dates
### to get past data requires correct date input 
### only futures data after 20190628 is available

In [4]:
acc.get_expired_futures_dates()


expiration date: 1225 | Date December 2020
expiration date: 0925 | Date September 2020
expiration date: 0626 | Date June 2020
expiration date: 20200327 | Date March 2020
expiration date: 20191227 | Date December 2019
expiration date: 20190927 | Date September 2019
expiration date: 20190628 | Date June 2019
expiration date: 20190329 | Date March 2019


## specific expired futures data 
### here, I'm using Ether for demo


In [27]:
eth_hist_futures = acc.get_expired_futures_OHCL(market = 'ETH-20190927', resolution = 3600)
print(eth_hist_futures)

Adding 1648 candles with start time 2019-09-26 23:00:00
Adding 1 candles with start time 2019-07-20 08:00:00
                      timestamp    open    high     low   close     volume  \
0     2019-07-20T12:00:00+00:00  227.76  228.38  227.41  227.62    0.00000   
1     2019-07-20T13:00:00+00:00  227.62  230.33  225.93  227.06    0.00000   
2     2019-07-20T14:00:00+00:00  227.06  228.91  227.01  228.80    0.00000   
3     2019-07-20T15:00:00+00:00  228.80  232.33  228.01  231.90    0.00000   
4     2019-07-20T16:00:00+00:00  231.90  236.97  231.49  235.48    0.00000   
...                         ...     ...     ...     ...     ...        ...   
1643  2019-09-26T23:00:00+00:00  166.77  167.73  165.82  166.10    0.00000   
1644  2019-09-27T00:00:00+00:00  166.10  166.71  164.19  164.34  198.98362   
1645  2019-09-27T01:00:00+00:00  164.26  165.89  163.83  165.84  339.72944   
1646  2019-09-27T02:00:00+00:00  165.81  165.95  164.33  165.05    0.00000   
1647  2019-09-27T03:00:00+00:00  